internal/services/disks/disk_pool_iscsi_target_resource.go

```diff
@@ -0,0 +1,289 @@
+				validation.StringLenBetween(5, 223),
+				validation.StringMatch(
+					regexp.MustCompile(`[a-z\d.\-]*[a-z\d]$`),
+					"The iSCSI target name can only contain lowercase letters, numbers, periods, or hyphens.",
+				),
+			),
+		},
+		"acl_mode": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice(
+				[]string{string(iscsitargets.IscsiTargetAclModeDynamic)},
+				false,
+			),
+		},
+		"disks_pool_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: iscsitargets.ValidateDiskPoolID,
+		},
+		"lun": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.All(
+							validation.StringMatch(
+								regexp.MustCompile(`^[\dA-Za-z-_.]+[\dA-Za-z]$`),
+								"supported characters include [0-9A-Za-z-_.]; name should end with an alphanumeric character"),
+							validation.StringLenBetween(1, 90),
+						),
+					},
+					"managed_disk_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: computeValidate.ManagedDiskID,
+					},
+					"number": {
+						Type:     pluginsdk.TypeInt,
+						Computed: true,
+					},
+				},
+			},
+		},
+		"target_iqn": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.IQN,
+		},
+	}
+}
+
+func (d DisksPoolIscsiTargetResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"endpoints": {
+			Type: pluginsdk.TypeList,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+			Computed: true,
+		},
+		"port": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+	}
+}
+
+func (d DisksPoolIscsiTargetResource) ModelObject() interface{} {
+	return &DisksPoolIscsiTargetResource{}
+}
+
+func (d DisksPoolIscsiTargetResource) ResourceType() string {
+	return "azurerm_disk_pool_iscsi_target"
+}
+
+func (d DisksPoolIscsiTargetResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			m := DiskPoolIscsiTargetModel{}
+			err := metadata.Decode(&m)
+			if err != nil {
+				return err
+			}
+			poolId, err := iscsitargets.ParseDiskPoolID(m.DisksPoolId)
+			if err != nil {
+				return err
+			}
+			if poolId.SubscriptionId != metadata.Client.Account.SubscriptionId {
+				return fmt.Errorf("Disk Pool subscription id %q is different from provider's subscription", poolId.SubscriptionId)
+			}
+			id := iscsitargets.NewIscsiTargetID(poolId.SubscriptionId, poolId.ResourceGroupName, poolId.DiskPoolName, m.Name)
+			client := metadata.Client.Disks.DisksPoolIscsiTargetClient
+			locks.ByID(poolId.ID())
+			defer locks.UnlockByID(poolId.ID())
```

Comment:
could we add whitespace throughout this PR for readability 
```suggestion
			m := DiskPoolIscsiTargetModel{}
			err := metadata.Decode(&m)
			if err != nil {
				return err
			}
			
			poolId, err := iscsitargets.ParseDiskPoolID(m.DisksPoolId)
			if err != nil {
				return err
			}
			if poolId.SubscriptionId != metadata.Client.Account.SubscriptionId {
				return fmt.Errorf("Disk Pool subscription id %q is different from provider's subscription", poolId.SubscriptionId)
			}
			
			id := iscsitargets.NewIscsiTargetID(poolId.SubscriptionId, poolId.ResourceGroupName, poolId.DiskPoolName, m.Name)
			client := metadata.Client.Disks.DisksPoolIscsiTargetClient
			locks.ByID(poolId.ID())
			defer locks.UnlockByID(poolId.ID())
```

etc